pkg/uninstall/uninstall.go

```diff
@@ -97,3 +135,35 @@ func (obj *uninstaller) createWatcher(ctx context.Context, pkg ctrlpkg.Package)
 		return nil, fmt.Errorf("unexpected object kind: %v", pkg.GroupVersionKind().Kind)
 	}
 }
+
+func (uninstaller *uninstaller) deleteNamespace(ctx context.Context, namespace string) {
+	clientset := cliutils.KubernetesClient(ctx)
+	err := clientset.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "❌ error deleting namespace: %v\n", err)
+		cliutils.ExitWithError()
+	}
+}
+
+func validateNamespaceDeletion(ctx context.Context, pkg ctrlpkg.Package) bool {
+	client := cliutils.PackageClient(ctx)
+	namespace := pkg.GetNamespace()
+	var pkgs v1alpha1.PackageList
+	if err := client.Packages(namespace).GetAll(ctx, &pkgs); err != nil {
+		fmt.Fprintf(os.Stderr, "❌ error listing packages in namespace: %v\n", err)
+		cliutils.ExitWithError()
+	}
+
+	var namespacePackages = make([]string, 0, len(pkgs.Items))
+	for _, pkg := range pkgs.Items {
+		namespacePackages = append(namespacePackages, pkg.Name)
+	}
+
+	if len(namespacePackages) > 1 {
+		fmt.Printf("\n❌ Namespace %s cannot be deleted because it contains other packages: %v\n",
+			namespace, strings.Join(namespacePackages, ", "))
+		return false
+	}
+
+	return true
+}
```

Comment:
I think this function should be refactored to return an error in case the namespace is not safe to delete instead of printing a message to stdout